Update Speculative Rendering (#7662)
GoogleChrome/web.dev: src/site/content/en/blog/speculative-prerendering/index.md

  </figure>
</div>

## Prerendering from the address bar

Starting with Chrome 101, prerendering will gradually be applied to the most likely navigation when a user triggers autocomplete suggestions by interacting with Chrome’s address bar. We are starting with Chrome for Android but intend to bring this feature to the other platforms we support in follow-up milestones.

Before this launch, Chrome used [No-State Prefetch](#prerender-implementation-using-no-state-prefetch) to prefetch critical resources for the most likely navigations from the address bar. Prerendering takes things a bit further by rendering the page, which includes constructing the DOM tree and executing scripts.

> While this is a browser feature, it can have observable effects from the point of view of the prerendered websites. For instance, APIs that may cause unexpected surprises are delayed until the user navigates to the prerendered web page (e.g., text to speech from an invisible website). 

Reviewer: ```suggestion
While this is a browser feature, it can have observable effects from the point of view of the prerendered websites. For instance, APIs that may cause unexpected surprises are delayed until the user navigates to the prerendered web page (for example, text to speech from an invisible website). 
```

> In case you wonder about the potential impact on server resource usage, prerendering from the [Omnibox](https://www.chromium.org/user-experience/omnibox/) only triggers on high-confidence suggestions. So, in practice, the server load increase should be negligible since the user **will** navigate to a vast majority of the prerendered web pages. That said, it’s easy to distinguish navigation requests triggered for prerendering or prefetch needs versus regular navigation requests. Look for the <code>[Sec-Purpose: prefetch](https://chromestatus.com/feature/6247959677108224)</code>; prerender HTTP header or its predecessor <code>Purpose: prefetch</code> sent by Chrome 101 and earlier versions. You can then decide to reduce server load in peak season further or prevent prerendering/prefetching if these cause issues for your website. Concretely, the server can cancel the prerendering by responding with an HTTP error or without a response body, as described below:
> 

Reviewer: ```suggestion
The potential impact on server resource usage is low, as prerendering from the [Omnibox](https://www.chromium.org/user-experience/omnibox/) only triggers on high-confidence suggestions. The server load increase should be negligible since the user **will** navigate to a vast majority of the prerendered web pages. Navigation requests triggered for prerendering or prefetch needs are distinguishable from regular navigation requests. Look for the <code>[Sec-Purpose: prefetch](https://chromestatus.com/feature/6247959677108224)</code>; prerender HTTP header or its predecessor <code>Purpose: prefetch</code> sent by Chrome 101 and earlier versions. You can then decide to reduce server load in peak season further or prevent prerendering or prefetching if these cause issues for your website. The server can cancel the prerendering by responding with an HTTP error or without a response body, as described below:

```
